### DISCUSSION TITLE
Jog key and hard limit switch problem

### DISCUSSION THREAD

**Discussion by jal123sin** [terjeio/ioSender#316] [Q&A] [answered]
I use the iosender edge 2.0.43.P9 for lathe control. As gcode display on the 3D veiw area, the iosender program will hangup when I pressed the jog key.  And, when I enable the hard limits, the limit switch will be triggered automatically, even I unpluged all the limit cables. Therefore, I must disable the hard limits. But the functioning of the home switch is good.

**Comment by terjeio** [comment]
Whcih controller? Version? If grblHAL then what is the response from the $LEV command? It shows the last events received from the switches. If a limit switch is shown to be triggered even without any connected it migh be due to EMI.

**Comment by jal123sin** [comment]
The last event sometimes will show [LASTEVENTS:S,ZY,,,],.
![P1](https://github.com/terjeio/ioSender/assets/136598254/cb55ad67-0b21-4335-a7dc-7f1b1d178837)

**Comment by jal123sin** [comment, marked as answer]
I follow the event listed, then I invert the Y setting, the limit alarm disappear at once. The machine is running well now. Thank you.
